ra/ra.go

```diff
@@ -242,22 +242,37 @@ type certificateRequestEvent struct {
 // registration-based overrides are necessary.
 const noRegistrationID = -1
 
-func (ra *RegistrationAuthorityImpl) checkRegistrationLimit(ctx context.Context, ip net.IP) error {
-	limit := ra.rlPolicies.RegistrationsPerIP()
+func (ra *RegistrationAuthorityImpl) checkRegistrationLimits(ctx context.Context, ip net.IP) error {
```

Comment:
It looks like we could remove a lot of duplication by creating a function that takes a rate limit policy and an SA RPC function, and calls the given RPC to check against the given policy. What do you think?